src/python/pants/backend/codegen/protobuf/buf.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import List, cast
+
+from pants.core.util_rules.external_tool import TemplatedExternalTool
+from pants.engine.platform import Platform
+
+
+class Buf(TemplatedExternalTool):
```

Comment:
Nit, a few of us have started being more explicit with names like calling this `BufSubsystem`. I've found it makes it more clear what is a Target, Field, Subsystem, some custom type, etc.